src/Encoding/FilteredStream.php

```diff
@@ -57,8 +57,20 @@
      */
     public function __construct(StreamInterface $stream, $readFilterOptions = null, $writeFilterOptions = null)
     {
-        $this->readFilterCallback = Filter\fun($this->readFilter(), $readFilterOptions);
-        $this->writeFilterCallback = Filter\fun($this->writeFilter(), $writeFilterOptions);
+        switch (func_num_args()) {
```

Comment:
if we pass explicit null to the constructor, this will fail right? what if i want write filter options but no read filter options? we could instead do:

```
if ($readFilterOptions) {
    $this->readFilterCallback = Filter\fun($this->readFilter(), $readFilterOptions);
} else {
    $this->readFilterCallback = Filter\fun($this->readFilter());
}
and the same for write filter
```